// eslint-config-expo/flat/default.js
const coreConfig = require('eslint-config-expo/flat/utils/core.js');
const expoConfig = require('eslint-config-expo/flat/utils/expo.js');
const reactConfig = require('eslint-config-expo/flat/utils/react.js');
const typescriptConfig = require('eslint-config-expo/flat/utils/typescript.js');
const { allExtensions } = require('eslint-config-expo/flat/utils/extensions.js');
const { defineConfig } = require('eslint/config');
const globals = require('eslint-config-expo/node_modules/globals');

module.exports = defineConfig([
  ...coreConfig,
  ...typescriptConfig,
  ...reactConfig,
  ...expoConfig,
  {
    settings: {
      'import/extensions': allExtensions,
      'import/resolver': {
        node: { extensions: allExtensions },
      },
    },
    languageOptions: {
      globals: {
        ...globals.browser,
        __DEV__: 'readonly',
        ErrorUtils: false,
        FormData: false,
        XMLHttpRequest: false,
        alert: false,
        cancelAnimationFrame: false,
        cancelIdleCallback: false,
        clearImmediate: false,
        fetch: false,
        navigator: false,
        process: false,
        requestAnimationFrame: false,
        requestIdleCallback: false,
        setImmediate: false,
        window: false,
        'shared-node-browser': true,
      },
    },
  },
  {
    files: ['*.web.*'],
  },
]);
